tests/models/whisper/test_modeling_whisper.py

```diff
@@ -0,0 +1,964 @@
+        labels = processed_inputs.labels.to(torch_device)
+
+        logits = model(
+            input_features,
+            decoder_input_ids=labels,
+            output_hidden_states=False,
+            output_attentions=False,
+            use_cache=False,
+        )
+
+        logits = logits.last_hidden_state @ model.decoder.embed_tokens.weight.T
+
+        # fmt: off
+        EXPECTED_LOGITS = torch.tensor(
+            [
+                2.1382, 0.9381, 4.4671, 3.5589, 2.4022, 3.8577, -0.6521, 2.5472,
+                1.8301, 1.9957, 2.3432, 1.4678, 0.5459, 2.2597, 1.5179, 2.5357,
+                1.1624, 0.6194, 1.0757, 1.8259, 2.4076, 1.6601, 2.3503, 1.3376,
+                1.9891, 1.8635, 3.8931, 5.3699, 4.4772, 3.9184
+            ]
+        )
+        # fmt: on
+
+        self.assertTrue(torch.allclose(logits[0, 0, :30].cpu(), EXPECTED_LOGITS, atol=1e-4))
+
+    def test_tiny_en_generation(self):
+
+        torch_device = "cpu"
+        set_seed(0)
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny.en")
+        model.to(torch_device)
+        model.config.decoder_start_token_id = 50257
+
+        input_speech = self._load_datasamples(1)
+        feaure_extractor = WhisperFeatureExtractor()
+
+        input_features = feaure_extractor(raw_speech=input_speech, return_tensors="pt").input_features.to(torch_device)
+
+        tokenizer = WhisperTokenizer.from_pretrained("openai/whisper-tiny.en")
+        generated_ids = model.generate(input_features, num_beams=5, forced_bos_token_id=50362)
+        transcript = tokenizer.batch_decode(generated_ids)[0]
+
+        EXPECTED_TRANSCRIPT = (
+            "<|startoftranscript|> <|notimestamps|>  Mr. Quilter is the apostle of the middle"
+            " classes, and we are glad to"
+        )
+        self.assertEqual(transcript, EXPECTED_TRANSCRIPT)
+
+    @slow
+    def test_tiny_generation(self):
+
+        torch_device = "cpu"
+        set_seed(0)
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model.to(torch_device)
+
+        input_speech = self._load_datasamples(1)
+        feaure_extractor = WhisperFeatureExtractor()
+
+        input_features = feaure_extractor(raw_speech=input_speech, return_tensors="pt").input_features.to(torch_device)
+
+        tokenizer = WhisperTokenizer.from_pretrained("openai/whisper-large")
+
+        decoder_input_ids = torch.tensor([[50258]]).long()
+        generated_ids = model.generate(input_features, num_beams=5, decoder_input_ids=decoder_input_ids)
+        transcript = tokenizer.decode(generated_ids[0])
+
+        EXPECTED_TRANSCRIPT = (
+            "<|startoftranscript|> <|en|> <|transcribe|> <|notimestamps|>  Mr. Quilter is the apostle of the middle"
+            " classes and we are glad"
+        )
+        self.assertEqual(transcript, EXPECTED_TRANSCRIPT)
+
+    @slow
+    def test_large_generation(self):
```

Comment:
Let's also add a multi-lingual test here and one for speech translation if you manage to get it working

Reply:
Okay, gonna add a multi-lingual audio, and check speech translation